src/mina-signer/signature.ts

```diff
@@ -204,3 +215,90 @@ function verify(
     return false;
   }
 }
+
+// legacy signatures
+
+/**
+ * Same as {@link sign}, but using the "legacy" style of hash input packing.
+ */
+function signLegacy(
+  message: HashInputLegacy,
+  privateKey: PrivateKey,
+  networkId: NetworkId
+): Signature {
+  let publicKey = Group.scale(Group.generatorMina, privateKey);
+  let kPrime = deriveNonceLegacy(message, publicKey, privateKey, networkId);
+  if (Scalar.equal(kPrime, 0n)) throw Error('sign: derived nonce is 0');
+  let { x: rx, y: ry } = Group.scale(Group.generatorMina, kPrime);
+  let k = Field.isEven(ry) ? kPrime : Scalar.negate(kPrime);
+  let e = hashMessageLegacy(message, publicKey, rx, networkId);
+  let s = Scalar.add(k, Scalar.mul(e, privateKey));
+  return { r: rx, s };
+}
+
+/**
+ * Same as {@link verify}, but using the "legacy" style of hash input packing.
+ */
+function verifyLegacy(
```

Comment:
Same comment above. You might want to refactor to keep a single implementation of the signing and verifying algorithms.